### PR TITLE
Add adaptive options to ODEs

### PR DESCRIPTION
**Pull request by everettk**
The ODEs for adaptive optimizers (Adam, LaProp) have a simple structure that just divides either g3 or g1 by the gradient norm (and we use sqrt(risk) as a proxy for the grad norm). Instead of maintaining separate implementations for each adaptive optimizer, this makes the main ODE code support adaptive options.

### CODE REVIEW

File: power_law_rf/ode.py [hunk 1]
```diff
@@ -105,68 +120,110 @@ def inverse_3x3(omega):
...
+        return jnp.einsum('i,j->ij', Gamma, inputs.eigs_K)  # 3 x d
 
     def ode_update(carry, time):
         v, twice_risk = carry
         time_plus = jnp.exp(time + dt)
         time_plus_minus_one = time_plus - 1.0
-        omega = omega_approximate(time_plus_minus_one) if approximate else omega_full(time_plus_minus_one)
+        
+        # Use sqrt(risk) as proxy for gradient norm when adaptive is not None
+        grad_norm = jnp.sqrt(twice_risk / 2.0) if adaptive is not None else 1.0
```
Comment (elliotpaquette) [inline power_law_rf/ode.py]: grad_norm = jnp.sqrt(twice_risk)

Comment (everettk) [reply power_law_rf/ode.py]: Fixed, thanks!
